packages/leva/stories/input-options.stories.tsx

```diff
@@ -118,3 +118,89 @@ export const OnChange = () => {
     </>
   )
 }
+
+export const OnChangeWithRender = ({ transient }) => {
+  const ref = React.useRef<HTMLPreElement | null>(null)
+  const data = useControls({
+    color: {
+      value: '#f00',
+      onChange: (value) => {
+        ref.current.innerHTML = value
+      },
+      transient,
+    },
+  })
+
+  return (
+    <div style={{ padding: 20, margin: 20, border: '1px solid black' }}>
+      <pre>color data (should{transient ? ' not' : null} update)</pre>
+      <pre>{JSON.stringify(data, null, '  ')}</pre>
+      <pre>Transient Value (should also update)</pre>
+      <pre ref={ref}></pre>
+    </div>
+  )
+}
+OnChangeWithRender.args = {
+  transient: false,
+}
+
+export const OnChangeStartOnChangeEnd = () => {
+  const [isEditing, setIsEditing] = React.useState(0)
+  const data = useControls({
+    string: {
+      value: 'foobars',
+      onChangeStart: () => {
+        setIsEditing((i) => i + 1)
+      },
+      onChangeEnd: () => {
+        setIsEditing((i) => i - 1)
+      },
+    },
+    number: {
+      value: 1,
+      onChangeStart: () => {
+        setIsEditing((i) => i + 1)
+      },
+      onChangeEnd: () => {
+        setIsEditing((i) => i - 1)
+      },
+    },
+    numberSlider: {
+      value: 1,
+      onChangeStart: () => {
+        setIsEditing((i) => i + 1)
+      },
+      onChangeEnd: () => {
+        setIsEditing((i) => i - 1)
+      },
+      min: 0,
+      max: 10,
+    },
+    vector2d: {
+      value: [1, 1],
+      onChangeStart: () => {
+        setIsEditing((i) => i + 1)
+      },
+      onChangeEnd: () => {
+        setIsEditing((i) => i - 1)
+      },
+    },
+    vector3d: {
+      value: [1, 1, 1],
+      onChangeStart: () => {
+        setIsEditing((i) => i + 1)
+      },
+      onChangeEnd: () => {
+        setIsEditing((i) => i - 1)
+      },
+    },
+  })
+
+  return (
+    <div style={{ padding: 20, margin: 20, border: '1px solid black' }}>
+      <pre>Value</pre>
+      <pre>{JSON.stringify(data, null, '  ')}</pre>
+      <pre>Is Editing? {isEditing === 0 ? 'No' : 'Yes'}</pre>
+    </div>
+  )
+}
```

Comment:
This showcases how the API can be used to keep track of whether something is currently being edited in user-land. The user can then also decide whether he wants to use `useRef` or `useState` for doing so.